driver/pace/driver/run.py

```diff
@@ -508,10 +586,22 @@ def step_all(self):
             self._step(timestep=self.config.timestep.total_seconds())
             time += self.config.timestep
             self.diagnostics.store(time=time, state=self.state)
+        self.performance_config.collect_performance(
+            self.performance_config.timestep_timer,
+            self.performance_config.times_per_step,
+            self.performance_config.hits_per_step,
+        )
 
     def _step(self, timestep: float):
-        self._step_dynamics(timestep=timestep)
-        self._step_physics(timestep=timestep)
+        with self.performance_config.timestep_timer.clock("mainloop"):
+            self._step_dynamics(timestep=timestep)
+            self._step_physics(timestep=timestep)
+        self.performance_config.collect_performance(
+            self.performance_config.timestep_timer,
+            self.performance_config.times_per_step,
+            self.performance_config.hits_per_step,
+        )
+        self.performance_config.timestep_timer.reset()
```

Comment:
Should `reset` be called automatically inside `collect_performance`? I see you call it here but not in `step_all`.

Reply:
The last one wasn't reset since it's not used again, but it is not a problem if it gets reset either. Moved `reset` to `collect_performance`